### ISSUE TITLE
`hss_reserve_signature` should fail if requested reserve value exceeds maximum count for the given key

### ISSUE THREAD

**Issue by MarkLuk** [cisco/hash-sigs#26]
The current function `hss_reserve_signature` implementation doesn't fail if requested reserve value increment, exceeds the key counter above the maximum value. It increments it to the maximum value without any user indication.

For robustness, this function shall return an error indication in this scenario.

For example, indicate error here via  `info->error_code`:
https://github.com/cisco/hash-sigs/blob/73310fdf7f83013adba5b393264955b9ca5f54cf/hss_reserve.c#L168

**Comment by sfluhrer** [comment]
Hmmm, the question really is "what does this API actually ask the code to do?"  Is it:
- Make sure that we have K signatures available (which won't fail, even if we can't update the state in the meantime); or
- Optimize the writes to the state so that we can skip the next K.
When I wrote this code, I was thinking about the second.  However, the autoreserve feature (which I added latter) really does it better; in addition, the code will fail if you try to reserve a huge number of signatures (which doesn't make sense in the second meaning).

So, I'm leaning towards accepting this change...
